lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -139,53 +140,58 @@ def notify_incoming_sms_handler(event: dict, context: any):
     if not valid_event(event):
-        logger.critical("Invalid event: %s", event)
+        logger.error("Invalid event: %s", event)
 
         # Push the message to the dead letter queue, and return 200 to have the message removed from feeder queue.
         push_to_sqs(event, False)
         return create_response(200)
 
-    # SQS events should contain "Records".  This is checked in the call above to valid_event.
+    # SQS events should contain "Records", and each record should contain "body".
+    # This is checked in the call above to valid_event.
     for record in event["Records"]:
-        logger.info("Processing an SQS inbound_sms record . . .")
-
-        # SQS event records should contain "body".  This is checked in the call above to valid_event.
+        # The body should be stringified JSON.
         record_body = record["body"]
 
         try:
             record_body = json.loads(record_body)
-            inbound_sms = json.loads(record_body.get("Message", ''))
+            inbound_sms = json.loads(record_body.get("Message"))
         except (json.decoder.JSONDecodeError, TypeError) as e:
-            # Deadletter
-            push_to_sqs(record_body, False)
-            logger.error("Malformed record body.")
+            # Dead letter
             logger.exception(e)
+            logger.error("Malformed record body: %s", record_body)
+            push_to_sqs(record_body, False)
             continue
 
         if not valid_message_body(inbound_sms):
-            logger.critical("The record's message body is invalid: %s", inbound_sms)
+            logger.error("The record's message body is invalid: %s", inbound_sms)
 
-            # Push the record to the dead letter queue, and return 200 to have the message removed from feeder queue.
+            # Push the record to the dead letter queue, and return 200 to have the message removed from the feeder queue.
             push_to_sqs(record_body, False)
+            if len(event["Records"]) > 1:
+                # TODO - Remove this block as part of #1014.  The problem is that the handler returns if any record has
+                # an invalid message.  It's not a problem now because we only expect one record.
+                logger.warning("Multiple records might not be processed correctly until batching is implemented.  See #1014.")
             return create_response(200)
+        # Else, the message has all the required fields.
 
-        # destinationNumber is the number the end user responded to (the 10DLC pinpoint number).
-        two_way_record = two_way_sms_table_dict.get(inbound_sms.get("destinationNumber"))
+        # destinationNumber is the 10DLC Pinpoint number to which the end user responded.
+        two_way_record = two_way_sms_table_dict.get(inbound_sms["destinationNumber"])
         if two_way_record is None:
-            # Deadletter
+            # Dead letter
+            logger.error("Unable to find a two_way_record for %s.", inbound_sms["destinationNumber"])
```

Comment:
I know you check for it but I really think no logging messages should have unsafe checks. If the code elsewhere changes, this has the opportunity to be a bug down the road. Using `.get` means it will never throw an exception (as no logger should).

Reply:
Fair enough.  I will change it.

Reply:
Thanks! Sorry for being stubborn here. I really think it's safer.